### DISCUSSION TITLE
more shit on killsay

### DISCUSSION THREAD

**Discussion by AXDWT** [Fedoraware/Fedoraware#987] [Ideas]
add more shit on killsay. for example:
%name%
%class%
%selfname%
%lastperson%
%randomname%
%randomfriendname%
and should be good ffs

**Comment by ghost** [comment]
name and class already exist.

**Comment by johnathon-walnut** [comment]
Name is %victim% since that's what nullcore uses so I figured it would be worth using the same

**Comment by ghost** [comment]
honestly I was gonna add this but I got to local name and realised this is almost entirely pointless (also what does %lastperson% even refer to).

**Comment by johnathon-walnut** [comment]
Maybe the previous person you killed?

I could envision like "Heh.. at least you put up more of a fight than %lastvictim%....."

**Comment by SeasonalFerret** [comment]
Used to have a lot of fun with this stuff on other cheats. Some more suggestions:
-Enemy health before you shot them `%priorhealth%`
-Damage dealt to kill them %dmg%)
-Your class `%myclass%`
-The name of their primary weapon (is this possible to grab?) `%wpnname%`
-The team the person is on `%team%`
~~-The name of the person who assisted you (or the actual killer if you get an assist) `%assisted%`~~ This wouldn't be useful unless we had different killsays for assist kills.
-The body part that you hit (if hitscan + not a shotgun) `%bodypart%`
-The number of times you've killed that person in the session `%killnumber%`
-The total number of their deaths `%deathstat%`
-The total number of your kills `%mykillstat%`
-Randomly generated IP address lmao `%fakeip%`


Examples:
"I dealt `%dmg%` damage to `%name%` and kill them! `%killnumber%` - 0"
"Wow `%name%` you're.. actually a really bad `%class%`. Try playing `%myclass%`?"
"I've killed `%name%` `%killnumber%` times, bringing their total deaths to `%deathstat%`"
"`%mykillstat%` noobs pwned and counting"
"lmao get %bodypart%-shot"
"%name% - %fakeip%"





Alternatively, if this isn't already added- How about a separate killsay for critical hits? I don't really like using killsay, but having a killsay for crits for something like "sorry for the crit %name%" would be nice


And a separate killsay toggle for people flagged as cheaters would be neat. Would be cool to call them out on being bad without annoying anyone else. Imagine spamming chat against one person.
For example, setting cheater-killsay to "%killnumber%" would just have you auto-type an incrementing number when you kill a cheater. Basically doing what all other cheaters do automatically lmao



Also, idea for two forms of delay-
How about a delay between the kill and the say, that way it looks like we're typing their names manually instead of automating it? This would be pretty nice.
Second, how about a "minimum delay" between two killsays, that way we don't spam chat? For example, setting this to 1 minute would give a random killsay every kill with a limit to 1 per minute, that way it doesn't spam chat and we don't get kicked for spamming.

**Comment by johnathon-walnut** [comment]
I think an easier option for crits would just to be to start your line with %iscrit% or smth

**Comment by SeasonalFerret** [comment]
> I think an easier option for crits would just to be to start your line with %iscrit% or smth


That would be really useful, actaully.
In that case, can we also have %isassist%, %isdomination%, %isrevenge%

You might want make it so that f the file _doesn't_ have %iscrit%, just run regular killsay on crits as well
If there's an %isrevenge% killsay, we run that on revenge kills. Otherwise, we treat revengekills as normal kills. Same with iscrit, isassist, etc.

Setting up a medic bot with %isassist% spams would be funny af

**Comment by SeasonalFerret** [comment]
Also fyi here's an old shit-talk list from like 2016
https://hastebin.com/omatiboneb.sql
